presto-spark-base/src/main/java/com/facebook/presto/spark/execution/HttpNativeExecutionTaskInfoFetcher.java

```diff
@@ -144,7 +144,7 @@ public void stop()
     public Optional<TaskInfo> getTaskInfo()
             throws RuntimeException
     {
-        if (scheduledFuture != null && scheduledFuture.isCancelled()) {
+        if (scheduledFuture != null && scheduledFuture.isCancelled() && lastException.get() != null) {
```

Comment:
What is this change about? Seems unrelated.

Reply:
While this class was being used by `NativeExecutionOperator`, it was assumed that `getTaskInfo` will always be called while task is running. 
The way the new `TaskExecutorFactory` is implemented, that is not the case. We `stop()` the task first and then do a final query to get `TaskInfo` for statistics.

I can change that implementation to extract `TaskInfo` before stopping, but it seemed like improving this check was a better approach. 

Not leaning strongly either way, let me know if you have a preference